ghcide/src/Development/IDE/Core/Rules.hs

```diff
@@ -1055,20 +1056,28 @@ writeHiFileAction hsc hiFile = do
         resetInterfaceStore extras $ toNormalizedFilePath' targetPath
         writeHiFile hsc hiFile
 
+data RulesConfig = RulesConfig
+    { -- | Disable import cycle checking for improved performance in large codebases
+      checkForImportCycles :: Bool
+    -- | Disable TH for improved performance in large codebases
+    , enableTemplateHaskell :: Bool
```

Comment:
The Eval plugin will also not work if this is `False`. Is that correct?

Reply:
Quite the opposite, this setting will not work if the Eval plugin is enabled, since it will redefine the `NeedsCompilation` rule and override this setting.